arch/arc/core/dsp/swap_dsp_macros.h

```diff
@@ -260,3 +260,14 @@ dsp_skip_load :
 #endif
 agu_skip_load :
 .endm
+
+.macro _dsp_extension_probe
+	lr	 r0, [_ARC_V2_DSP_BUILD]
+	bmsk r0, r0, 7
+	breq r0, 0, skip_dsp_setup
+#ifdef CONFIG_ARC_DSP_TURNED_OFF
+	mov	 r0, 0 /* DSP_CTRL_DISABLED_ALL */
+	sr	 r0, [_ARC_V2_DSP_CTRL]
+#endif
+skip_dsp_setup :
+.endm
```

Comment:
```suggestion
.macro _dsp_extension_probe
#ifdef CONFIG_ARC_DSP_TURNED_OFF
	lr	 r0, [_ARC_V2_DSP_BUILD]
	bmsk r0, r0, 7
	breq r0, 0, skip_dsp_setup
	mov	 r0, 0 /* DSP_CTRL_DISABLED_ALL */
	sr	 r0, [_ARC_V2_DSP_CTRL]
skip_dsp_setup :
#endif
.endm
```

Reply:
Probably even drop BCR check - as we assume that platform should set `CONFIG_ARC_HAS_DSP` proper way.